frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -426,55 +197,178 @@ export function applyStreamlitThemeTemplateLayout(
     },
   }
 
-  // cant use merge. use assign because plotly already has properties defined.
-  assign(layout, streamlitTheme)
+  merge(layout, streamlitTheme)
 }
 
 /**
- * Applies the colorscale, colors for unique graphs, and discrete coloring
- * for in general through assigning properties in spec.data
- * @param data - spec.data
+ * Replace the colors that we are using from streamlit_plotly_theme.py.
+ * This is done so that we change colors based on the background color
+ * as the backend has no idea of the background color.
+ * @param spec the spec that we want to update
+ * @param theme
+ * @param elTheme element.theme
```

Comment:
nit: I was a bit confused about the naming before reading the comment, since `elementTheme` isn't too long of a name either, it would make it slightly more readable.